test/integration/text_layer_spec.mjs

```diff
@@ -0,0 +1,243 @@
+              );
+          })
+        );
+      });
+
+      it("doesn't jump when hovering on an empty area (multi-page)", async () => {
+        await Promise.all(
+          pages.map(async ([browserName, page]) => {
+            const scrollTarget = await getSpanRectFromText(
+              page,
+              1,
+              "Unlike method-based dynamic compilers, our dynamic com-"
+            );
+            await page.evaluate(top => {
+              document.getElementById("viewerContainer").scrollTop = top;
+            }, scrollTarget.y - 50);
+
+            const [
+              positionStartPage1,
+              positionEndPage1,
+              positionStartPage2,
+              positionEndPage2,
+            ] = await Promise.all([
+              getSpanRectFromText(
+                page,
+                1,
+                "Each compiled trace covers one path through the program with"
+              ).then(middlePosition),
+              getSpanRectFromText(
+                page,
+                1,
+                "or that the same types will occur in subsequent loop iterations."
+              ).then(middlePosition),
+              getSpanRectFromText(
+                page,
+                2,
+                "Hence, recording and compiling a trace"
+              ).then(middleBottomPosition),
+              getSpanRectFromText(
+                page,
+                2,
+                "cache. Alternatively, the VM could simply stop tracing, and give up"
+              ).then(belowEndPosition),
+            ]);
+
+            await page.mouse.move(positionStartPage1.x, positionStartPage1.y);
+            await page.mouse.down();
+
+            await moveInSteps(page, positionStartPage1, positionEndPage1, 20);
+            await moveInSteps(page, positionEndPage1, positionStartPage2, 20);
+
+            expect(await getSelectedText(page))
+              .withContext(`In ${browserName}, first selection`)
+              .toMatch(
+                browserName === "chrome"
+                  ? /^e path through the program .*Hence, recording an$/s
+                  : /^ path through the program .*Hence, recording an$/s
+              );
+
+            await moveInSteps(page, positionStartPage2, positionEndPage2, 20);
+            await page.mouse.up();
+
+            expect(await getSelectedText(page))
+              .withContext(`In ${browserName}, second selection`)
+              .toMatch(
+                browserName === "chrome"
+                  ? /^e path through.*Hence, recording and .* tracing, and give up$/s
+                  : /^ path through.*Hence, recording and .* tracing, and give $/s
+              );
+          })
+        );
+      });
+    });
+
+    describe("using selection carets", () => {
+      let browser;
+      let page;
+
+      beforeAll(async () => {
+        browser = await startBrowser({
+          browserName: "firefox",
```

Comment:
My understanding is that the test suite can either be run in Firefox or Firefox+Chrome modes, and that `it`/`describe` blocks are run only once and not once per browser (iteration across the browsers happens in `loadAndWait`, which is run _inside_ the tests). I will add a comment, but this test is running only for Firefox because there is no concept of "running the test suite for Chrome".